### PR TITLE
IDE-3711 adapt target name synchronized changing node name

### PR DESCRIPTION
**Pull request by JoyeLuo**

### CODE REVIEW

File: enterprise/plugins/com.liferay.ide.kaleo.core/src/com/liferay/ide/kaleo/core/model/internal/NodeNameChangeListener.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+import org.eclipse.sapphire.Element;
+import org.eclipse.sapphire.ElementList;
+import org.eclipse.sapphire.FilteredListener;
+import org.eclipse.sapphire.Property;
+import org.eclipse.sapphire.PropertyContentEvent;
+
+/**
+ * @author Haoyi Sun
+ */
+public class NodeNameChangeListener extends FilteredListener<PropertyContentEvent> {
```
Comment (jtydhr88) [inline enterprise/plugins/com.liferay.ide.kaleo.core/src/com/liferay/ide/kaleo/core/model/internal/NodeNameChangeListener.java]: Just named NodeListener

File: enterprise/plugins/com.liferay.ide.kaleo.core/src/com/liferay/ide/kaleo/core/model/internal/NodeNameChangeListener.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+
+	@Override
+	protected void handleTypedEvent(PropertyContentEvent event) {
+		Element elementByModify = event.property().element();
```
Comment (jtydhr88) [inline enterprise/plugins/com.liferay.ide.kaleo.core/src/com/liferay/ide/kaleo/core/model/internal/NodeNameChangeListener.java]: just named **element**

**Comment by jtydhr88** [comment]
@haoyisun lets talk more about it

File: enterprise/plugins/com.liferay.ide.kaleo.core/src/com/liferay/ide/kaleo/core/model/internal/NodeNameChangeListener.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+				}
+			}
+		}
+
+		for (Transition transition : allTransitions) {
+			String transitionTargetName = transition.getTarget().toString();
+
+			if (transitionTargetName.equals(nodeOldName)) {
+				if (nodeNewName != null) {
+					transition.setTarget(nodeNewName);
```
Comment (jtydhr88) [inline enterprise/plugins/com.liferay.ide.kaleo.core/src/com/liferay/ide/kaleo/core/model/internal/NodeNameChangeListener.java]: need to remove the pair from map by the event.property()

File: enterprise/plugins/com.liferay.ide.kaleo.core/src/com/liferay/ide/kaleo/core/model/internal/NodeNameChangeListener.java [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+	protected void handleTypedEvent(PropertyContentEvent event) {
+		Element elementByModify = event.property().element();
+
+		WorkflowDefinition workflow = elementByModify.adapt(WorkflowDefinition.class);
+
+		Map<String, String> eventInfo = new HashMap<>();
+
+		event.fillTracingInfo(eventInfo);
+
+		String nodeOldName = eventInfo.get("before");
```
Comment (jtydhr88) [inline enterprise/plugins/com.liferay.ide.kaleo.core/src/com/liferay/ide/kaleo/core/model/internal/NodeNameChangeListener.java]: use oldName

File: enterprise/plugins/com.liferay.ide.kaleo.core/src/com/liferay/ide/kaleo/core/model/internal/NodeNameChangeListener.java [hunk 5]
```diff
@@ -0,0 +1,125 @@
...
+
+		if (nodeOldName == null) {
+			nodeOldName = _target_relate_node.get(event.property());
+		}
+
+		String nodeNewName = eventInfo.get("after");
```
Comment (jtydhr88) [inline enterprise/plugins/com.liferay.ide.kaleo.core/src/com/liferay/ide/kaleo/core/model/internal/NodeNameChangeListener.java]: use newName

File: enterprise/plugins/com.liferay.ide.kaleo.core/src/com/liferay/ide/kaleo/core/model/internal/NodeNameChangeListener.java [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+				ElementList<Transition> taskTransitions = task.getTransitions();
+
+				for (Transition transition : taskTransitions) {
+					allTransitions.add(transition);
+				}
+			}
+		}
+
+		for (Transition transition : allTransitions) {
+			String transitionTargetName = transition.getTarget().toString();
```
Comment (jtydhr88) [inline enterprise/plugins/com.liferay.ide.kaleo.core/src/com/liferay/ide/kaleo/core/model/internal/NodeNameChangeListener.java]: use targetName

File: enterprise/plugins/com.liferay.ide.kaleo.core/src/com/liferay/ide/kaleo/core/model/internal/NodeNameChangeListener.java [hunk 7]
```diff
@@ -0,0 +1,125 @@
...
+import org.eclipse.sapphire.Property;
+import org.eclipse.sapphire.PropertyContentEvent;
+
+/**
+ * @author Haoyi Sun
+ */
+public class NodeNameChangeListener extends FilteredListener<PropertyContentEvent> {
+
+	@Override
+	protected void handleTypedEvent(PropertyContentEvent event) {
```
Comment (jtydhr88) [inline enterprise/plugins/com.liferay.ide.kaleo.core/src/com/liferay/ide/kaleo/core/model/internal/NodeNameChangeListener.java]: add
Property property = event.property();

File: enterprise/plugins/com.liferay.ide.kaleo.core/src/com/liferay/ide/kaleo/core/model/internal/NodeNameChangeListener.java [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+					transition.setTarget(nodeNewName);
+				}
+				else {
+					_target_relate_node.put(event.property(), nodeOldName);
+				}
+			}
+		}
+	}
+
+	private static final Map<Property, String> _target_relate_node = new HashMap<>();
```
Comment (jtydhr88) [inline enterprise/plugins/com.liferay.ide.kaleo.core/src/com/liferay/ide/kaleo/core/model/internal/NodeNameChangeListener.java]: use notNullNodeNames

**Comment by JoyeLuo** [comment]
will resend

**Pull request closed without merging**